src/main/java/com/novocode/junit/JUnitTask.java

```diff
@@ -0,0 +1,100 @@
+package com.novocode.junit;
+
+import junit.framework.TestCase;
+import org.junit.experimental.categories.Categories;
+import org.junit.runner.Description;
+import org.junit.runner.JUnitCore;
+import org.junit.runner.Request;
+import org.junit.runner.RunWith;
+import sbt.testing.*;
+
+import java.lang.annotation.Annotation;
+import java.util.*;
+
+final class JUnitTask implements Task {
+  private static final Fingerprint JUNIT_FP = new JUnitFingerprint();
+
+  private final JUnitRunner runner;
+  private final RunSettings settings;
+  private final TaskDef taskDef;
+
+  public JUnitTask(JUnitRunner runner, RunSettings settings, TaskDef taskDef) {
+    this.runner = runner;
+    this.settings = settings;
+    this.taskDef = taskDef;
+  }
+
+  @Override
+  public String[] tags() {
+    return new String[0];  // no tags yet
+  }
+
+  @Override
+  public TaskDef taskDef() { return taskDef; }
+
+  @Override
+  public Task[] execute(EventHandler eventHandler, Logger[] loggers) {
+    Fingerprint fingerprint = taskDef.fingerprint();
+    String testClassName = taskDef.fullyQualifiedName();
+    Description taskDescription = Description.createSuiteDescription(testClassName);
+    RichLogger logger = new RichLogger(loggers, settings, testClassName);
+    EventDispatcher ed = new EventDispatcher(logger, eventHandler, settings, fingerprint, taskDescription, runner.runStatistics);
+    JUnitCore ju = new JUnitCore();
+    ju.addListener(ed);
+    if (runner.runListener != null) ju.addListener(runner.runListener);
+
+    Map<String, Object> oldprops = settings.overrideSystemProperties();
```

Comment:
Can you add a comment explaining why this is necessary? I'm not exactly sure what it's doing, but it seems a little hackish and I wonder if there isn't a better way

Reply:
It's more than a little hackish but this is all old code that I only moved around. No actual changes here.